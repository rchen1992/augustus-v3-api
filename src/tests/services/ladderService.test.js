const createLadderService = require('@services/ladderService');
const mockLadderRepo = require('@tests/mockRepos/ladder');

describe('Ladder Service', () => {
    test('should be able to calculate user rank', async () => {
        const service = createLadderService(mockLadderRepo);
        const ladderId = 2;

        let user1Rank = await service.getUserRank(ladderId, '1');
        expect(user1Rank).toBe(4);

        let user2Rank = await service.getUserRank(ladderId, '2');
        expect(user2Rank).toBe(2);

        let user4Rank = await service.getUserRank(ladderId, '4');
        expect(user4Rank).toBe(1);

        let user5Rank = await service.getUserRank(ladderId, '5');
        expect(user5Rank).toBe(3);
    });
});
