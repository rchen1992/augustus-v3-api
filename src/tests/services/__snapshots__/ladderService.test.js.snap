// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ladder Service should be able to get ladder users sorted by rank desc 1`] = `
Array [
  Object {
    "created_at": "2018-09-27 20:12:13",
    "ladder_id": 2,
    "rating": 220,
    "rating_delta": 20,
    "updated_at": "2018-05-28 10:49:38",
    "user": Object {
      "avatar_url": "https://robohash.org/molestiaeinciduntlaboriosam.bmp?size=50x50&set=set1",
      "created_at": "2019-01-01 16:47:52",
      "email": "gbanat3@typepad.com",
      "updated_at": "2018-04-20 21:04:19",
      "user_id": "4",
      "user_name": "etappor3",
    },
    "user_id": "4",
  },
  Object {
    "created_at": "2018-09-27 20:12:13",
    "ladder_id": 2,
    "rating": 200,
    "rating_delta": 20,
    "updated_at": "2018-05-28 10:49:38",
    "user": Object {
      "avatar_url": "https://robohash.org/nesciuntnobismagni.png?size=50x50&set=set1",
      "created_at": "2018-08-01 19:26:23",
      "email": "swoodyatt1@storify.com",
      "updated_at": "2018-07-14 22:48:36",
      "user_id": "2",
      "user_name": "csaye1",
    },
    "user_id": "2",
  },
  Object {
    "created_at": "2018-09-27 20:12:13",
    "ladder_id": 2,
    "rating": 120,
    "rating_delta": 20,
    "updated_at": "2018-05-28 10:49:38",
    "user": Object {
      "avatar_url": "https://robohash.org/voluptatumundemodi.bmp?size=50x50&set=set1",
      "created_at": "2018-08-16 17:09:23",
      "email": "sdennistoun4@wikimedia.org",
      "updated_at": "2018-08-18 01:01:35",
      "user_id": "5",
      "user_name": "selcott4",
    },
    "user_id": "5",
  },
  Object {
    "created_at": "2018-07-24 06:11:55",
    "ladder_id": 2,
    "rating": 100,
    "rating_delta": 30,
    "updated_at": "2018-11-30 11:49:27",
    "user": Object {
      "avatar_url": "https://robohash.org/solutaeaut.bmp?size=50x50&set=set1",
      "created_at": "2018-09-25 03:08:51",
      "email": "mjurczak0@epa.gov",
      "updated_at": "2018-07-12 05:22:33",
      "user_id": "1",
      "user_name": "cblanko0",
    },
    "user_id": "1",
  },
]
`;
